src/goto-instrument/contracts/utils.cpp

```diff
@@ -163,3 +167,118 @@ void disable_pointer_checks(source_locationt &source_location)
   source_location.add_pragma("disable:pointer-primitive-check");
   source_location.add_pragma("disable:pointer-overflow-check");
 }
+
+bool is_parameter_assign(
+  const goto_programt::targett &instruction_it,
+  namespacet &ns)
+{
+  optionalt<symbol_exprt> sym = {};
+
+  // extract symbol
+  if(instruction_it->is_assign())
+  {
+    const auto &lhs = instruction_it->assign_lhs();
+    if(can_cast_expr<symbol_exprt>(lhs))
+      sym = to_symbol_expr(lhs);
+  }
+
+  // check condition
+  if(sym.has_value())
+    return ns.lookup(sym.value().get_identifier()).is_parameter;
+
+  return false;
```

Comment:
```suggestion
  if(instruction_it->is_assign() && can_cast_expr<symbol_exprt>(instruction_it->assign_lhs()))
  {
    // Extract symbol from lhs
    optionalt<symbol_exprt> lhs_sym = to_symbol_expr(instruction_it->assign_lhs());
    // Check whether is parameter
    if(lhs_sym.has_value())
      return ns.lookup(lhs_sym.value().get_identifier()).is_parameter;
  }

  return false;
```

Reply:
On top of what Felipe suggested: there's no need for the `optionalt` anymore.